### PR TITLE
fix(session): disable http keep alive

### PR DESCRIPTION
**Pull request by cardyok**

### CODE REVIEW

File: internal/session/session.go [hunk 1]
```diff
@@ -168,22 +167,7 @@ func (s *Session) startWriter() {
...
-				}).DialContext,
-				ForceAttemptHTTP2:     true,
-				MaxIdleConns:          10,
-				IdleConnTimeout:       30 * time.Second,
-				TLSHandshakeTimeout:   10 * time.Second,
-				ExpectContinueTimeout: 1 * time.Second,
+				DisableKeepAlives: true,
 			},
 		}
 		resp, err := client.Do(req)
```
Comment (xiang90) [inline internal/session/session.go:173]: is there a timeout on this do func?

File: internal/session/session.go [hunk 2]
```diff
@@ -251,22 +235,7 @@ func (s *Session) startReader() {
...
```
Comment (xiang90) [inline internal/session/session.go:241]: it seems there is no timeout here as well...

File: internal/session/session.go (hunk 1 above)
Comment (xiang90) [reply internal/session/session.go:173]: writecloser needs to somehow get passed in here.

**Comment by xiang90** [comment]
basically:

1. create a http client every single time... do not reuse any http client.
2. on the reader side, it reads every x seconds. if there is nothing, then the read side of the tcp connection may be broken. it should close both the reader and the writer.
3. on the writer side, it writes every x seconds. if it fails to write, then the write side of the tcp connection may be broken. it should close both the reader and the writer.

**Comment by xiang90** [comment]
also do extensive logging everywhere.

**Review by gyuho — APPROVED** [review]
lgtm

**Pull request merged**